server/game/GameActions/ConditionalGameAction.ts

```diff
@@ -0,0 +1,66 @@
+import { GameAction, GameActionProperties } from './GameAction';
+import AbilityContext = require('../AbilityContext');
+
+export interface ConditionalActionProperties extends GameActionProperties {
+    gameAction: GameAction;
+    condition: boolean | ((context: AbilityContext) => boolean);
```

Comment:
Would this ever be boolean?  I think it should probably always be a function....